### PR TITLE
Add Support for Fastify Backend Framework

### LINKED ISSUE #1991: Add Support for Fastify instead Express for Backend Framework
**Is your feature request related to a problem? Please describe.**

 Fastify is known for its LTS for the Typescript community Fastify has LTS and has shown to be significantly faster than Express and has the same features as Express support. Fastify is a modern web framework highly focused on providing the best developer experience with the least overhead and a powerful plugin architecture. It is one of the fastest modern web frameworks and also can be a good alternative to Express because it can process requests 20% faster than ExpressJS. Fastify is also faster than Express in terms of response time and request throughput. 

**Describe the solution you'd like**

Migrate current backend framework support of express to fastify without affecting any other functionalities 

**Describe alternatives you've considered**

N/A

**Approach to be followed (optional)**

1. Migrate the current backend framework i.e.. express to support fastify in `develop-userTypeFix` branch
2. Remove all packages related to express and if needed,  restore with the modern alternative packages.

**Additional context**
N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by varshith257**
**What kind of change does this PR introduce?**

**Issue Number:**

- Fixes #1991

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

Will be added soon

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

Yes. The impact could be a result of the server setup.

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by varshith257** [comment]
@xoldd PTAL

**Comment by Huy1996** [comment]
@varshith257 Please fix the failing test.

**Comment by varshith257** [comment]
@Cioppolo14 I will be inactive this coming week due to exams. I will be continue the progress of my assigned issues and PRs from the coming weekend.

**Comment by xoldd** [comment]
The PR is incomplete, it has not accommodated all the changes needed for a graceful transition.

**Comment by varshith257** [comment]
I had a doubts with them soo I made a PR to let you know the progress

**Comment by varshith257** [comment]
@xoldd See my review

**Comment by xoldd** [comment]
@varshith257 Like I had said in the issue tracker, simply reimplementing it in fastify while keeping 1:1 compatibility is wasteful. You're still trying to use express specific logic/packages with fastify instead of doing it the fastify way. Other than that here are a few more things that need fixing:-

1. `src/libraries/logger.ts`:- needs to be migrated to pino which comes built-in with fastify
2. `src/libraries/requestContext.ts`:- needs to be migrated to i18next-middleware
3. `src/libraries/requestTracing.ts`:- useless, needs to be removed

File: package.json [hunk 1]
```diff
@@ -43,14 +43,23 @@
   },
   "homepage": "https://github.com/PalisadoesFoundation/talawa-api#readme",
   "dependencies": {
-    "@apollo/server": "^4.10.0",
+    "@apollo/server": "^4.10.2",
+    "@as-integrations/fastify": "^2.1.1",
```
Comment (xoldd) [inline package.json:47]: just checked and this doesn't support the latest version of fastify

File: src/app.ts [hunk 2]
```diff
@@ -56,49 +78,80 @@ i18n.configure({
...
+  }
 );
-app.use(mongoSanitize());
-app.use(cors());
 
-app.use("/images", (req, res, next) => {
-  res.setHeader("Cross-Origin-Resource-Policy", "cross-origin");
-  next();
+fastifyApp.addHook('preHandler', (request, reply, done) => {
+  mongoSanitize();
```
Comment (xoldd) [inline src/app.ts:94]: this is not needed

File: src/app.ts [hunk 3]
```diff
@@ -56,49 +78,80 @@ i18n.configure({
...
+
+
+fastifyApp.addHook('onRequest', (request, reply, done) => {
+  reply.header("Cross-Origin-Resource-Policy", "cross-origin");
+  done();
+});
+
+
+fastifyApp.addHook('preHandler', (request, reply, done) => {
+  graphqlUploadExpress();
```
Comment (xoldd) [inline src/app.ts:106]: this is not needed

**Comment by varshith257** [comment]
@palisadoes There are getting issues with adding fastify support packages with current apollo server  

image

And also the reason for this is https://github.com/PalisadoesFoundation/talawa-api/pull/2194#discussion_r1566403287

**Comment by varshith257** [comment]
And also same have been stated here: https://github.com/PalisadoesFoundation/talawa-api/issues/2028#issuecomment-2004782880. As in the Slack discussion, you are thinking of holding this Yoga migration. Waiting for your suggestion here

**Comment by palisadoes** [comment]
Yes we should wait. I'll close this and the issue.

**Pull request closed without merging**